### PR TITLE
StaveTie -- move rendering logic out of draw

### PR DESCRIPTION
**Pull request by mscuthbert**
Moving the rendering logic (where to start, where to end, etc.) out of the draw() method for StaveTie so that it can be tested separately from the visual tests.  (And hopefully expanded soon).

Added ability to configure the cutoff point for where a tie gets classified as a short tie and what the effects of that are.

Added a visual test (w/ some application logic tests) of a more standard use of ties: two chords that have the same pitches, and with two different ties between the same chords representing different directions.  (Surprisingly, this most standard use of ties was not being tested).
Screenshot 2024-03-08 at 12 21 20

Still learning the factory system.  :-)

Let me know if changes like this are appreciated, neutral, or -1.  

Preparatory work for fixing the ties w/ lyrics problem in #182

### CODE REVIEW

**Comment by rvilarl** [comment]
Changes like that are appreciated from my side. @ronyeh what do you think?

**Comment by ronyeh** [comment]
Haven't had a chance to review this yet, but just wanted to say that I appreciate any and all contributions to improving code & documention quality.

I like that you are adding comments as you go. It'll definitely help everyone who uses VexFlow (including ourselves)!

**Comment by mscuthbert** [comment]
Great!  Then I'll continue to add comments as I investigate.  Sometimes the comments might be in weird places -- for instance, if I'm looking to see if Beams affect formatter to lead to poor behavior, and I conclude that it didn't, but I had to figure out something about beaming along the way I'll leave any comments I made there.

**Comment by ronyeh** [comment]
Even if comments are fuzzy or incomplete, or even if they are of the "Huh?" type, I think it will help. When we all can admit to not understanding a part of VexFlow, we can work together to improve the code clarity & quality (or even fix any subtle bugs that confusing code can introduce).

File: tests/annotation_tests.ts [hunk 1]
```diff
@@ -7,31 +7,27 @@
...
+import { Registry } from '../src/registry';
+import { ContextBuilder } from '../src/renderer';
+import { Stave } from '../src/stave';
+import { StaveNote, StaveNoteStruct } from '../src/stavenote';
+import { Stem } from '../src/stem';
+import { TabNote, TabNoteStruct } from '../src/tabnote';
+import { TabStave } from '../src/tabstave';
+import { Tickable } from '../src/tickable';
+import { Vibrato } from '../src/vibrato';
+import { Voice } from '../src/voice';
```
Comment (ronyeh) [inline tests/annotation_tests.ts:30]: My copy of annotation_tests.ts already has these eslint fixes.

Can you fetch the most recent revision and rebase off of main? Maybe these changes will disappear?

Comment (ronyeh) [reply tests/annotation_tests.ts:30]: https://github.com/vexflow/vexflow/blob/main/tests/annotation_tests.ts

Comment (mscuthbert) [reply tests/annotation_tests.ts]: Changes disappeared with the last real commit and pulling "main".

**Pull request merged**